bundles/org.openhab.core.automation.module.script.rulesupport/src/main/java/org/openhab/core/automation/module/script/rulesupport/loader/ScriptFileWatcher.java

```diff
@@ -46,32 +46,26 @@
 import org.openhab.core.service.ReadyMarkerFilter;
 import org.openhab.core.service.ReadyService;
 import org.openhab.core.service.StartLevelService;
-import org.osgi.service.component.annotations.Activate;
-import org.osgi.service.component.annotations.Component;
-import org.osgi.service.component.annotations.Deactivate;
-import org.osgi.service.component.annotations.Reference;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * The {@link ScriptFileWatcher} watches the jsr223 directory for files. If a new/modified file is detected, the script
+ * The {@link ScriptFileWatcher} watches a directory for files. If a new/modified file is detected, the script
  * is read and passed to the {@link ScriptEngineManager}.
  *
  * @author Simon Merschjohann - Initial contribution
  * @author Kai Kreuzer - improved logging and removed thread pool
- * @author Jonathan Gilbert - added dependency tracking & per-script start levels
+ * @author Jonathan Gilbert - added dependency tracking & per-script start levels; made reusable
  */
-@Component(immediate = true)
 public class ScriptFileWatcher extends AbstractWatchService implements ReadyService.ReadyTracker,
```

Comment:
Should this be an abstract class now?

Reply:
It could be, although I left it as concrete as technically code could (usefully) instantiate an instance without extending it. Happy to make it abstract if you'd prefer.

Reply:
No, not really necessary.